compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -112,20 +148,28 @@ resolveTileSwizzlingType(RankedTensorType packedType,
   for (auto i : encodingInfo.permutation) {
     perm.push_back(i + srcRank);
   }
+  // unrolling would require tiled inner shape. Eventually it would be like:
+  // From: [ABo, ACCo, ABi, ACCi] into:
+  // [ABo, ACCo, AB1, ABunroll, AB2, ACC1, ACC2]
+  // where: ABi = AB1 * ABunroll * AB2
+  // and: ACCi = ACC1 * ACC2
 
-  // TODO(lialan): Support unrolling and K-interleaving.
   SmallVector<int64_t> intrinsicVectorShape = encodingInfo.innerTileShapes;
-  auto iT1 = intrinsicVectorShape[0];
-  auto iT2 = intrinsicVectorShape[1];
-  auto oT1 = encodingInfo.innerTileSizes[0] / iT1;
-  auto oT2 = encodingInfo.innerTileSizes[1] / iT2;
+  auto packedShape = packedType.getShape();
+  auto packedRank = packedType.getRank();
+  auto AB1 = packedShape[packedRank - 2] / encodingInfo.innerTileSizes[0];
+  auto AB2 = intrinsicVectorShape[0];
+  auto ABunroll = encodingInfo.innerTileSizes[0] / AB2;
+  auto ACC1 = packedShape[packedRank - 1] / intrinsicVectorShape[1];
+  auto ACC2 = intrinsicVectorShape[1];
```

Comment:
nit: we follow camelCase naming style in IREE.

```suggestion
  auto ab1 = packedShape[packedRank - 2] / encodingInfo.innerTileSizes[0];
  auto ab2 = intrinsicVectorShape[0];
  auto abUnroll = encodingInfo.innerTileSizes[0] / ab2;
  auto acc1 = packedShape[packedRank - 1] / intrinsicVectorShape[1];
  auto acc2 = intrinsicVectorShape[1];
```